src/PluginClientFactoryInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Http\Client\Common;
+
+use Http\Client\HttpAsyncClient;
+use Http\Client\HttpClient;
+
+/**
+ * Create a PluginClient.
+ *
+ * @author Fabien Bourigault <bourigaultfabien@gmail.com>
+ */
+interface PluginClientFactoryInterface
+{
+    /**
+     * @param HttpClient|HttpAsyncClient $client
+     * @param Plugin[]                   $plugins
+     * @param array                      $options
+     *
+     * @see PluginClient constructor for $options details.
```

Comment:
You are correct. I had in mind that one could create ANY `PluginClient` with this. But the `PluginClientFactoryInterface` creates an instance of the concrete class `PluginClient`. That is a huge different from our other `*FactoryInterface`s. They create an instance of an interface.